xarray/core/groupby.py

```diff
@@ -45,6 +45,9 @@
 
     GroupKey = Any
 
+    T_GroupIndicesListInt = list[list[int]]
+    T_GroupIndices = Union[T_GroupIndicesListInt, list[slice], np.ndarray]
```

Comment:
Would be nice to reduce this to 1 consistent type.

Reply:
The `np.ndarray` one is potentially removable after https://github.com/pydata/xarray/issues/2157